### PR TITLE
WI #86 #317 Improve support of SYMBOLIC CHARACTERS

### LINKED ISSUE #317: Bug with special names paragraph
TC parser doesn't recognize this kind of special names paragraph:

``` cobol
       special-names.
                  decimal-point  comma
                  SYMBOLIC CHARACTERS LOW-VAL IS 1
                        HEX-15  IS 22
                        HI-VAL  IS 256
                  CLASS NAMECHAR IS
             'A' THRU 'I'
             'J' THRU 'R'
             'S' THRU 'Z'
```

### PR DESCRIPTION
**Pull request by fm-117**
Fixes #317

Improve support for SYMBOLIC CHARACTERS (issue #86), parser now recognizes _symbolic-character-reference_ in different contexts.

### CODE REVIEW

File: TypeCobol.Test/Parser/Programs/Cobol85/SpecialNamesParagraph.rdz.cbl [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+       01 t1 PIC X(10).
+       01 t2 PIC X(10).
+       01 t3 PIC X(10).
+       01 c PIC X.
+       PROCEDURE DIVISION.
+           MOVE 'M' TO c
+           COPY CopyUsingSpecialNames.
+           GOBACK
+           .
+       END PROGRAM TCOMFL06.
```
Comment (rooksdo) [inline TypeCobol.Test/Parser/Programs/Cobol85/SpecialNamesParagraph.rdz.cbl:28]: I suggest the following code sample to test that `SYMBOLIC CHARACTERS` is well supported (provided the execution output)
```cob
      
       IDENTIFICATION DIVISION.
       PROGRAM-ID. TCOMFL06.
       ENVIRONMENT DIVISION.
       CONFIGURATION SECTION.
       special-names.
           decimal-point comma
           SYMBOLIC CHARACTERS LOW-VAL IS 1
                               HEX-15  IS 22
                               HI-VAL  IS 256
           CLASS NAMECHAR IS 'A' THRU 'I'
                             'J' THRU 'R'
                             'S' THRU 'Z'
           .
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       01 var1 PIC X(10).
       01 var2 PIC X(10).
       01 var3 PIC X(10).
       01 c    PIC X.
       PROCEDURE DIVISION.
      *   Test NAMECHAR
           MOVE 'M'      TO c
           IF c IS NAMECHAR
              DISPLAY "TEST NAMECHAR OK"
           ELSE
              DISPLAY "TEST NAMECHAR FAILED"
           END-IF
      *   Test LOW-VAL
           MOVE 1        TO var1
           IF var1 EQUAL LOW-VAL
              DISPLAY "TEST LOW-VAL OK"
           ELSE
              DISPLAY "TEST LOW-VAL FAILED"
           END-IF
      *   Test HEX-15
           MOVE 22       TO var2
           IF var2 EQUAL HEX-15
              DISPLAY "TEST HEX-15 OK"
           ELSE
              DISPLAY "TEST HEX-15 FAILED"
           END-IF
      *   Test HI-VAL
           MOVE 256      TO var3
           IF var3 EQUAL HI-VAL
              DISPLAY "TEST HI-VAL OK"
           ELSE
              DISPLAY "TEST HI-VAL FAILED"
           END-IF
           GOBACK
           .
```

**Review by rooksdo — CHANGES_REQUESTED** [review]

File: TypeCobol.Test/Parser/Programs/Cobol85/SpecialNamesParagraph.rdz.cbl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+       01 var1 PIC X(10) VALUE LOW-VAL.
+       01 var2 PIC X(10) VALUE HEX-15.
+       01 var3 PIC X(10) VALUE HI-VAL.
+       01 t1 PIC X(10).
+       01 t2 PIC X(10).
+       01 t3 PIC X(10).
+       01 c PIC X.
+       PROCEDURE DIVISION.
+           MOVE 'M' TO c
+           COPY CopyUsingSpecialNames.
```
[the remark this answers is not in the document]
Comment (fm-117) [reply TypeCobol.Test/Parser/Programs/Cobol85/SpecialNamesParagraph.rdz.cbl:25]: The idea of the test is to check that the symbolic characters defined in the program are recognized also when they are  used in a COPY.

[the remark this answers is not in the document]
Comment (rooksdo) [reply TypeCobol.Test/Parser/Programs/Cobol85/SpecialNamesParagraph.rdz.cbl:25]: OK

File: TypeCobol.Test/Parser/Programs/Cobol85/CopyUsingSpecialNames.cpy [hunk 3]
```diff
@@ -0,0 +1,6 @@
+           MOVE LOW-VAL TO t1
+           MOVE HEX-15  TO t2
+           MOVE HI-VAL  TO t3
+           IF c IS NAMECHAR
+              DISPLAY "ok"
```
[the remark this answers is not in the document]
Comment (rooksdo) [reply TypeCobol.Test/Parser/Programs/Cobol85/CopyUsingSpecialNames.cpy:5]: No sense.

**Review by rooksdo — APPROVED** [review]

**Review by smedilol — APPROVED** [review]

**Review by mayanje — APPROVED** [review]

**Pull request merged**